### PR TITLE
Update service_container.rst

### PR DESCRIPTION
**Pull request by l-vo**
From Symfony 4.0, there is no `public: false` in `_defaults` anymore. As I understand it's the default value (`true`) of the property `private` in [Definition class](https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/DependencyInjection/Definition.php) + the [ResolvePrivatesPass](
https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/DependencyInjection/Compiler/ResolvePrivatesPass.php) compiler pass that make services as private by default.

### CODE REVIEW

File: reference/dic_tags.rst [hunk 1]
```diff
@@ -96,9 +93,9 @@ services:
         use App\Lock\PostgresqlLock;
         use App\Lock\SqliteLock;
 
-        $container->register('app.mysql_lock', MysqlLock::class)->setPublic(false);
-        $container->register('app.postgresql_lock', PostgresqlLock::class)->setPublic(false);
-        $container->register('app.sqlite_lock', SqliteLock::class)->setPublic(false);
+        $container->register('app.mysql_lock', MysqlLock::class);
```
Comment (HeahDude) [inline reference/dic_tags.rst]: Looks like a left over here, we should use the `ContainerConfigurator` here. Would you mind changing it too, or we could do it in another PR.

File: reference/dic_tags.rst [hunk 2]
```diff
@@ -151,9 +148,9 @@ the generic ``app.lock`` service can be defined as follows:
...
```
Comment (HeahDude) [inline reference/dic_tags.rst]: Same as above.

File: security/custom_authentication_provider.rst [hunk 3]
```diff
@@ -435,14 +429,14 @@ to service ids that may not exist yet: ``App\Security\Authentication\Provider\Ws
 
         $container->register(WsseProvider::class)
             ->setArgument('$cachePool', new Reference('cache.app'))
-            ->setPublic(false);
+        ;
```
Comment (HeahDude) [inline security/custom_authentication_provider.rst]: Same.

File: serializer.rst [hunk 4]
```diff
@@ -118,7 +117,6 @@ properties and setters (``setXxx()``) to change properties:
         use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
 
         $container->register('get_set_method_normalizer', GetSetMethodNormalizer::class)
-            ->setPublic(false)
```
Comment (HeahDude) [inline serializer.rst]: Same here.

File: service_container/alias_private.rst [hunk 5]
```diff
@@ -63,7 +62,7 @@ You can also control the ``public`` option on a service-by-service basis:
             $services = $configurator->services();
 
             $services->set(Foo::class)
-                ->private();
+                ->setPublic();
```
Comment (HeahDude) [inline service_container/alias_private.rst]: `public()`

File: service_container/parent_services.rst [hunk 6]
```diff
@@ -248,8 +248,8 @@ the child class:
             $services->set(DoctrineUserRepository::class)
                 ->parent(BaseDoctrineRepository::class)
 
-                // overrides the public setting of the parent service
-                ->private()
+                // overrides the private setting of the parent service
+                ->setPublic()
```
Comment (HeahDude) [inline service_container/parent_services.rst]: `public()`

**Review by HeahDude — CHANGES_REQUESTED** [review]
Thank you for that PR, great work so far!

The PHP config needs more changes though, could you please take care of it too?

**Comment by l-vo** [comment]
@HeahDude ContainerConfiguration used instead of register calls 🙂

File: frontend/custom_version_strategy.rst [hunk 7]
```diff
@@ -149,7 +148,7 @@ After creating the strategy PHP class, register it as a Symfony service.
                     '%kernel.project_dir%/busters.json',
                     '%%s?version=%%s',
                 ]
-        )->setPublic(false);
```
Comment (HeahDude) [inline frontend/custom_version_strategy.rst]: PHP-DSL needed here

File: profiler/data_collector.rst [hunk 8]
```diff
@@ -268,7 +267,6 @@ to specify a tag that contains the template:
 
         $container
             ->autowire(RequestCollector::class)
-            ->setPublic(false)
```
Comment (HeahDude) [inline profiler/data_collector.rst]: Here too

File: reference/dic_tags.rst [hunk 9]
```diff
@@ -147,16 +151,25 @@ the generic ``app.lock`` service can be defined as follows:
...
+            $services = $configurator->services();
+
+            $services->set('app.mysql_lock', MysqlLock::class);
+            $services->set('app.postgresql_lock', PostgresqlLock::class);
+            $services->set('app.sqlite_lock', SqliteLock::class);
 
-        $container->register('app.lock')
-            ->addTag('auto_alias', ['format' => 'app.%database_type%_lock']);
+            $services
+                ->set('app.lock')
```
Comment (HeahDude) [inline reference/dic_tags.rst]: Please keep `set` on the first line:
```php
$services->set('app.lock')
    ->tag('auto_alias', ['format' => 'app.%database_type%_lock'])
;
```
(same below several times)	                 -

**Review by HeahDude — CHANGES_REQUESTED** [review]
A final round of comments, thanks again!

File: profiler/data_collector.rst (hunk 8 above)
Comment (l-vo) [reply profiler/data_collector.rst]: What do you call php-dsl ? The usage of ContainerConfigurator ?

File: frontend/custom_version_strategy.rst (hunk 7 above)
Comment (l-vo) [reply frontend/custom_version_strategy.rst]: What do you call php-dsl ? The usage of ContainerConfigurator ?

Comment (HeahDude) [reply frontend/custom_version_strategy.rst]: Yes :)

**Comment by l-vo** [comment]
@HeahDude Requested changes done :)

**Review by HeahDude — APPROVED** [review]
Looks very good, thanks a lot for your work on this!

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
These are very nice changes! Thanks Laurent.
